File: MosPosudit.Model/Requests/Tool/ToolUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Tool
{
    public class ToolUpdateRequest
    {
        [StringLength(100, ErrorMessage = "Tool name cannot be longer than 100 characters")]
        public string? Name { get; set; }

        [StringLength(500, ErrorMessage = "Description cannot be longer than 500 characters")]
        public string? Description { get; set; }

        public int? CategoryId { get; set; }

        [Range(0.01, 10000, ErrorMessage = "Daily rate must be between 0.01 and 10000")]
        public decimal? DailyRate { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Quantity must be at least 1")]
        public int? Quantity { get; set; }

        [Range(0, 10000, ErrorMessage = "Deposit amount must be between 0 and 10000")]
        public decimal? DepositAmount { get; set; }

        public bool? IsAvailable { get; set; }

        // Image as base64 string (for uploaded images)
        public string? ImageBase64 { get; set; }
    }
}

File: MosPosudit.UI/shared/lib/dtos/tool/tool_update_request.dart
class ToolUpdateRequestDto {
  final String? name;
  final String? description;
  final int? categoryId;
  final num? dailyRate;
  final int? quantity;
  final num? depositAmount;
  final bool? isAvailable;
  final String? imageBase64;

  ToolUpdateRequestDto({
    this.name,
    this.description,
    this.categoryId,
    this.dailyRate,
    this.quantity,
    this.depositAmount,
    this.isAvailable,
    this.imageBase64,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{};
    if (name != null) json['name'] = name;
    if (description != null) json['description'] = description;
    if (categoryId != null) json['categoryId'] = categoryId;
    if (dailyRate != null) json['dailyRate'] = dailyRate;
    if (quantity != null) json['quantity'] = quantity;
    if (depositAmount != null) json['depositAmount'] = depositAmount;
    if (isAvailable != null) json['isAvailable'] = isAvailable;
    if (imageBase64 != null) json['imageBase64'] = imageBase64;
    return json;
  }
}

